src/bootstrap-table.js

```diff
@@ -1463,10 +1466,11 @@
                 searchText: this.searchText,
                 sortName: this.options.sortName,
                 sortOrder: this.options.sortOrder
-            };
+            },
+            request;
 
-        if (!this.options.url) {
-            return;
+        if (!this.options.url && !this.options.ajax) {
+            throw new Error("Using Server requires either a `url` or a custom `ajax` method");
         }
```

Comment:
Notice that I've changed this condition from silently failing to send the ajax request to explicitly throwing an error.
This will give useful information if someone forget to set the `url` or `ajax` method after setting `sidePagination` to `server`, but might break existing code if someone has previously set the `sidePagination` to `server` without setting the `url` (not realizing no request is being sent).